backend/src/channels/repository/channel.repository.ts

```diff
@@ -0,0 +1,30 @@
+import { Channel } from '../../channels/entities/channel.entity';
+import { PageOptionsDto } from '../../common/dto/page-options.dto';
+import { QueryChannelDto } from '../dto/query-channel.dto';
+import { PageDto } from '../../common/dto/page.dto';
+import { PaginatedRepository } from '../../common/repository';
+import { EntityRepository } from 'typeorm';
+
+@EntityRepository(Channel)
+export default class ChannelRepository extends PaginatedRepository<Channel> {
+  public async findAllPaginated(
+    query?: QueryChannelDto,
+    pageOptions?: PageOptionsDto,
+  ): Promise<PageDto<Channel>> {
+    return this.findAllPaginated2(
+      pageOptions,
+      'channel',
+      'id',
+      (selectQueryBuilder) => {
+        if (query?.name)
+          selectQueryBuilder.where('channel.name = :name', {
+            username: query.name,
```

Comment:
```suggestion
            name: query.name,
```

To prevent internal service error when searching for a specific channel